src/h2olog/misc/gen_raw_tracer.py

```diff
@@ -436,16 +436,23 @@ def build_tracer(context, metadata):
 
   if fully_specified_probe_name == "h2o:_private_socket_lookup_flags":
     c += r"""
-#ifdef H2OLOG_SAMPLING_RATE_U32
   uint64_t flags = event._private_socket_lookup_flags.original_flags;
-  int skip_tracing = bpf_get_prandom_u32() > H2OLOG_SAMPLING_RATE_U32;
-  if (skip_tracing) {
-    flags |= H2O_EBPF_FLAGS_SKIP_TRACING_BIT;
+#ifdef H2OLOG_SAMPLING_RATE_U32
+  if ((flags & H2O_EBPF_FLAGS_SKIP_TRACING_BIT) == 0) {
+    if (bpf_get_prandom_u32() > H2OLOG_SAMPLING_RATE_U32)
+      flags |= H2O_EBPF_FLAGS_SKIP_TRACING_BIT;
+  }
+#endif
+#ifdef H2OLOG_IS_SAMPLING_ADDRESS
+  if ((flags & H2O_EBPF_FLAGS_SKIP_TRACING_BIT) == 0) {
+    if (!H2OLOG_IS_SAMPLING_ADDRESS(event._private_socket_lookup_flags.info.family,
+                                    event._private_socket_lookup_flags.info.remote.ip))
+      flags |= H2O_EBPF_FLAGS_SKIP_TRACING_BIT;
   }
+#endif
   int64_t ret = h2o_return.insert(&event._private_socket_lookup_flags.tid, &flags);
```

Comment:
@gfx It is my understanding that the existing code calls `h2o_return.insert` only when `H2OLOG_SAMPLING_RATE_U32` is defined. Is that intentional?

I think we should either:
* register `_private_socket_lookup_flags` only when sampling is necessary, or
* call `h2o_return.insert` only when the value of `flags` becomes different than `_private_socket_lookup_flags.original_flags`

I tend to believe that the former is better, but if we are to take that route, I think we'd better open a separate PR for pursuing that approach. WDYT?

Reply:
It's not intentional and I agree that we'll take the former, _register _private_socket_lookup_flags only when sampling is necessary_. I'm willing to make the pull request for that.

Reply:
Thanks. I think that'd be appreciated.